code/spaceship/solution/spaceship.sol.cpp

```diff
@@ -0,0 +1,33 @@
+#include <iostream>
+#include <cmath>
+#include <compare>
+
+struct Complex
+ {
+  double r, i ;
+  double norm() const { return std::sqrt(r*r+i*i) ; }
+  auto operator<=>( Complex const & other ) const
+   { return (norm()<=>other.norm()) ; }
```

Comment:
This is a bit of an overkill. Although optimization is not the topic here, let's have a `sqnorm` method returning `r*r+i*i` and use it in `operator<=>`